doc/Neighbor/Neighbor_Refresh_HLD.md

```diff
@@ -0,0 +1,321 @@
+ 8) "0"
+127.0.0.1:63792>
+```
+
+### 3.2.3 STATE DB
+Not applicable
+
+### 3.2.4 ASIC DB
+Not applicable
+
+### 3.2.5 COUNTER DB
+Not applicable
+
+### 3.2.6 Configuration save and restart
+Not applicable
+
+## 3.3 Switch State Service Design
+
+### 3.3.1 Orchestration Agent
+
+No Change in Orchangent / NeighOrch.
+
+### 3.3.2 NeighSyncd
+
+Neighsyncd is responsible for syncing the kernel neighbor table to the hardware via the APP_DB and OrchAgents. Neighsyncd listens on netlink events (RTM_NEWNEIGH, RTM_DELNEIGH) and creates/deletes NEIGH_TABLE entries in APP_DB.
+
+Existing functionality of neighsyncd is retained as it is. In addition to managing NEIGH_TABLE entries in APP_DB, neighsyncd will also add the details of the neighbor into NEIGH_REFRESH_TABLE.
+
+
+### 3.3.2 Nbrmgrd
+
+Nbrmgrd in addition to its existing responsibilities will now have to take care of the following Neighbor Refresh resonsibilities.
+
+- listens to NEIGH_REFRESH_TABLE and manage a local cache of neighbor entries
+- subscribe to redis-db interface CONFIG_DB tables to gather the data required to send the ARP refresh packets
+- subscribe to STATE_DB FDB table to manage the mac address table local cache
+- periodically refresh ARP/ND by sending ARP request pkt / NS pkt
+
+
+#### 3.3.2.1 Local Cache
+
+Neighbor Refresh Cache:
+
+- Add neighbor entries to cache when the entry is added to NEIGH_REFRESH_TABLE
+    - All Dynamically learned neighbor entries [ARP, ND (Global, LinkLocal)]
+    - All Static neighbor entries (MAC can be dynamic)
+- Following Entries will not be added to the neighbor cache
+    - Neighbors learned from “eth0” interface
+    - Neighbors learned from BGP/EVPN MAC/IP type-2 route 
+- Remove entries from cache when the entry is deleted from NEIGH_REFRESH_TABLE
+
+```
+v4/v6 Neighbors Cache [map]
+    
+    Key     : IP Address + InterfaceName [Phy/PortChannel/Vlan/Sag]
+
+    Value
+            : MAC Address
+            : IP address family
+            : State  (Reachable/Stale/Failed)
+            : Kernel Flags (For NTF_EXT_LEARNED)
+            : Timestamp (Entry creation/last refresh)
+```
+
+Interface Cache:
+
+- Interface IP address will be used in building the ARP (Sender IP address) and NS (Soure IP) packets
+- Subscribe to redis-db tables
+    - IP address 
+      - CONFIG_DB: INTERFACE, VLAN_INTERFACE, SAG_INTERFACE, PORTCHANNEL_INTERFACE
+
+```
+Interface Cache [Map]
+
+    Key     : Interface name
+    Value   : List of IP address configured
+```
+
+
+FDB Cache:
+
+- FDB Cache is to handle quick refresh/retry of neighbors whose mac address is removed admistratively or due to control protocol clear 
+- It has the list of neighbors(v4/v6) using the same mac on a VLAN interface. When the mac address is deleted, all the neighbors associated with the mac will be added to a fdb refresh queue.
+- Entries in FDB refresh queue will be refreshed faster than the usual neighbor refresh interval and will be removed from the queue after 10 retries.
```

Comment:
Do we report an error in this case i.e after 10 retries? during the STP topology settlement, 10 retries may not be sufficient, if we can present what is tested with this feature, that would act as the reference for the community.

Reply:
Yes there will be a syslog NOTICE/ERROR msg.

Sure, will verify with STP topology and update the recommended scale / guidelines.